packages/SwingSet/src/types.js

```diff
@@ -53,3 +53,39 @@
  *   exitVatWithFailure: unknown,
  * }} TerminationVatPowers
  */
+
+/*
+ * `['message', targetSlot, msg]`
+ * msg is `{ method, args, result }`
+ * `['notify', resolutions]`
+ * `['dropExports', vrefs]`
+ */
+
+/**
+ * @typedef {{
+ * method: string,
+ * args: SwingSetCapData,
+ * result?: string,
+ * }} Message
+ *
+ * @typedef { [tag: 'message', target: string, msg: Message]} VatDeliveryMessage
+ * @typedef { [tag: 'notify', resolutions: string[] ]} VatDeliveryNotify
+ * @typedef { [tag: 'dropExports', vrefs: string[] ]} VatDeliveryDropExports
```

Comment:
Yay! types! but...

Names such as `VatDeliveryDropExports` don't seem to get reused anywhere. I suspect the `VatDeliveryObject` and `VatSyscallObject` types would be more clear if the options were inlined.

minor style issue. not critical.

Reply:
Ah, I didn't know how to do that. Would it be something like:

```js
/**
 * @typedef { [tag: 'message', target: string, msg: Message] |
 *            [tag: 'notify', resolutions: string[]] |
 *            [tag: 'dropExports', vrefs: string[]] } VatDeliveryObject
 */
```
?

(note to self/chip, that `notify` signature isn't correct, I'll fix that)

At some point soon, we're going to have a function in liveslots which switches on `tag` and then calls one of three different functions. Would the typedefs on those three functions be a reason to retain the `VatDeliveryMessage`/etc subtypes?

Reply:
> Would it be something like: ... ?

yes.

> Would the typedefs on those three functions be a reason to retain the VatDeliveryMessage/etc subtypes?

yes.